### ISSUE TITLE
Troubles running the Hybrid-Handler due to dependencies

### ISSUE THREAD

**Issue by nilssass** [smash-transport/smash-vhlle-hybrid#49]
I tried to run the Hybrid-Handler locally on my macOS Sequoia 15.1.1. machine and ran into several hurdles:

- As already known the handler is not compatible with the newest version of numpy. On my machine, I am using numpy 1.23.5 which is too high already, so I need to set up a virtual environment.
- When trying to create a virtual environment to pip install numpy 1.11 I realized that numpy 1.11 is not compatible with the newer python versions, which is python 3.9.7 in my case. This means that I have to create a pyenv to create a venv to run the Hybrid-Handler.

I think it would be handy to allow for higher numpy versions, at least within numpy 1.xx such that at least the last step of creating a pyenv would become obsolete. Is there a certain reason why we limit the handler to ~1.11?

With that limitation I have to say that for me personally, the usage of the handler feels quite unhandy and not super intuitive.

**Comment by NGoetz** [comment]
Well it is a simple fix after all, the handler **is** compatible with higher numpy version, at least for me it runs smoothly. The requirements are just stricter than that. I think we should not enforce a singular numpy version.
Anyway, I wonder why we force 1.11, originally it was higher, but it was changed in 66c11cc . @AxelKrypton do you remember way?
Nevertheless, @nilssass , I don't think you need to panic, you can just use numpy 1.26, so it is not that unhandy after all.

**Comment by NGoetz** [comment]
This relates to #41 , where somebody wanted to use numpy 2. This I think would need to be tested and is probably a stretch. It is also not necessary as numpy 1 supports still all major Python releases.

**Comment by nilssass** [comment]
Thank you for the quick answer @NGoetz. Currently the problem is the error:

```
INTERNAL: Unexpected version value --- found when checking for Python numpy~=1.11 requirement.
           Please, contact developers.
```

when running `./Hybrid-handler do -c configs/config_AuAu_200.0_no_afterburner.yaml `. The message is originating from [here](https://github.com/smash-transport/smash-vhlle-hybrid/blob/8faa6f948c154c21554eb40f1c66bed0d4810a58/bash/system_requirements.bash#L322). Do you have any experience with it?

Of course I have made sure that numpy is installed and found when running a test program like
```
python3 -c "import numpy; print(numpy.__version__)"
```

**Comment by NGoetz** [comment]
Did you manually adapt the requirememts file to your version in your clone of your repo?

**Comment by nilssass** [comment]
Yes, I did. However it does not seem to get any version, since the message then still reads:

```
INTERNAL: Unexpected version value --- found when checking for Python numpy~=1.23 requirement.
           Please, contact developers.
```

**Comment by NGoetz** [comment]
I have not seen such an isdue before. Might be related to the _packaging_ package.

**Comment by nilssass** [comment]
Yes, I will try to have a look into it now. Thanks for the hint!

**Comment by nilssass** [comment]
Now, I understand the problem:

To check the required python package versions against the system installations, we use `python/test_requirement.py`. In [line 63](https://github.com/smash-transport/smash-vhlle-hybrid/blob/8faa6f948c154c21554eb40f1c66bed0d4810a58/python/test_requirement.py#L63) the script calls `pip freeze` to obtain the system informations about all installed python packages. However, if I manually run `pip freeze`, the output looks like this:

```
numpy @ file:///private/var/folders/sy/f16zz6x50xz3113nwtb9bvq00000gp/T/abs_8cgikbzvtd/croot/numpy_and_numpy_base_1672336184666/work
numpydoc @ file:///private/var/folders/sy/f16zz6x50xz3113nwtb9bvq00000gp/T/abs_adnyzxppoz/croot/numpydoc_1668085907252/work
...
pathspec==0.12.1
patsy==0.5.3
pep8==1.7.1
```

So, some packages contain their version numbers directly, while others contain the direct URL format. This format is not supported by test_requirement.py, which is the reason why the handler cannot access my numpy version. 

If I run `pip list` instead, all version numbers are represented as expected:
```
numpy                         1.23.5
numpydoc                      1.5.0
...
pathspec                      0.12.1
patsy                         0.5.3
pep8                          1.7.1
```

Of course, I can try to reinstall all python packages, but I think this would not be the ideal solution for other users. To me it looks like we should either allow for the URL format in `test_requirement.py` as well or we should use `pip list` instead of `pip freeze`. At least we should add a warning message to make the user aware, that the package could be found but not its version number due to an invalid format.

What do you think @NGoetz @AxelKrypton ?

**Comment by NGoetz** [comment]
We should probably change the command: https://stackoverflow.com/questions/61765502/pip-freeze-doesnt-show-package-version

**Comment by AxelKrypton** [comment]
Some comments from my side:

* Our requirement is `numpy~=1.11` and any `1.xx` version with `xx >= 11` is by semantic versioning expected to be compatible with version `1.11`. Therefore, I do not understand why @nilssass you say that version `1.23.5` is not complying with the requirement.
* Now, I think the statement about "not compatibility" above was misled by the problem you found, i.e. that the Handler is not always able to find the Python version of your `numpy` package. I searched a bit and it seems that [this SO answer](https://stackoverflow.com/a/62886215) sheds light on the problem. It is a `pip` problem I was not aware of. I think we should include in our codebase the workaround.
* Last comment about creating a `venv`. Reading your description, it sounded like the feeling was "Oh no, I have to create a `venv`..." - but I believe one should create virtual environments for Python projects as a totally natural thing. 🤔 

I will open a PR about this soon. It should be a one line change.

**Comment by NGoetz** [comment]
Closed by #50
